lib/hammer_cli_katello/id_name_options_validator.rb

```diff
@@ -0,0 +1,30 @@
+module IdNameOptionsValidator
+  # this is a module that requires:
+  # 1. either name or id is required
+  # 2. if name is supplied, some parent opt is also required
+  #
+  # Normally the parent will be organization as with products, content views,
+  # etc but this could also apply to repos for example whose names are unique
+  # per product
+
+  def validate_name_or_id_for(record_name, with_parent: 'organization', required: true)
+    validate_options do
+      parent_options = ["option_#{with_parent}_id", "option_#{with_parent}_name"]
```

Comment:
I'm not sure this is flexible enough for the default case of organization because you can (always?) specify organization by `['option_organization_id', 'option_organization_name', 'option_organization_label']`. Perhaps you could specify the argument like this:

```ruby
validate_name_or_id_for(record_name, with_parent: {'organization' => ['name', 'id', 'label']}, required: true)
```

and allow more than just one parent option. For example, creating a host requires a myriad of other options (i.e. domain, operatingsystem, architecture, partition_table), maybe specified in this method like this:

```ruby
validate_name_or_id_for(record_name, with_parent: {'domain' => ['name', 'id'], 'operatingsystem' => ['name', 'id'], 'architecture' => ['name', 'id'], 'partition_table' => ['name', 'id']}, required: true)
```

I'm not convinced of these changes, however.